### PR TITLE
Integrate typedoc to generate code documentation.

### PR DESCRIPTION
**Pull request by jerome-benoit**
Signed-off-by: Jérôme Benoit <jerome.benoit@sap.com>

## PR Checklist

- [x] Please add a description of your changes.
- [ ] We need your changes to come with unit tests in order to keep this project in quality and easy to maintain.
- [ ] Please add a link to the open issue or task that this pull request will resolve.
- [ ] Add a note to the changelog to indicate the change.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

If yes please indicate it and migration info into the CHANGELOG.md file.

---

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -50,3 +50,4 @@ jspm_packages
 .history/
 .scannerwork
 lib
+docs
```
Comment (pioardi) [inline .gitignore:53]: We already have a docs folder that contains our logo and should not be ignored right ?

File: README.md [hunk 2]
```diff
@@ -1,5 +1,5 @@
 <div align="center">
-<img src="./docs/logo.png" width="340px" height="266px"/>
+<img src="./images/logo.png" width="340px" height="266px"/>
```
Comment (pioardi) [inline README.md:2]: Oh ok I see it now

File: .gitignore (hunk 1 above)
Comment (jerome-benoit) [reply .gitignore:53]: docs/logo.png -> images/logo.png

**Review by pioardi — COMMENTED** [review]
Looks good, so we now generate our JSDoc but still not include into our git repo right ?

**Comment by Shinigami92** [comment]
You can drag and drop an image into a comment from a random issue in GitHub
This will generate a link for the image and this can be used from everywhere (like a Dropbox :laughing:)
So you could upload the logo and use the generated link in the readme instead of adding it to the repo
Downside would be that we havn't the raw versioned image, but it's not an svg so this should not be the raw original anyways

**Comment by pioardi** [comment]
> Downside would be that we havn't the raw versioned image, but it's not an svg so this should not be the raw original anyways

But why we should? We have a repo so we don' t need a link to an external issue or source

**Comment by jerome-benoit** [comment]
> Looks good, so we now generate our JSDoc but still not include into our git repo right ?

Not yet because I do not know how it's going to look like on GitHub. 
We must first test it before to finalize the typedoc integration. It's the first step, so others dev can have a look at the typedoc output, etc.

**Review by pioardi — APPROVED** [review]
This PR is good to go IMO

**Comment by Shinigami92** [comment]
We should generate the `docs` folder and commit it to GitHub
Then @pioardi needs to enable the GH pages feature for the `/docs` folder

File: .gitignore (hunk 1 above)
Comment (Shinigami92) [inline .gitignore:53]: Remove this one and commit the docs folder

**Review by Shinigami92 — APPROVED** [review]

**Comment by Shinigami92** [comment]
@pioardi Now you need to enable the docs as gh pages

Something like this on repo settings in the options section really far below
image

**Comment by pioardi** [comment]
Here we go https://pioardi.github.io/poolifier/

**Comment by Shinigami92** [comment]
@pioardi It's also possible to add a link to the repo that links to the page

1. Click on the settings cogs
2. Add the website link

example:

**Comment by pioardi** [comment]
Yep done bro.
@jerome-benoit  @Shinigami92  I think we are ready to release !!! what do you think ? Do you want to add any particular detail to our changelog before to release or you feel that we can move on ?

**Comment by Shinigami92** [comment]
@pioardi If you release, do not forget to remove/replace the changelogs `not released yet`

**Pull request merged**